pyccel/ast/headers.py

```diff
@@ -284,6 +284,10 @@ def build_argument(var_name, dc):
 
             order = None
             shape = None
+
+            if rank and precision == -1:
+                precision = default_precision[dtype]
```

Comment:
@aihya @azouiten I just wanted to point out that this syntax may also be used in pyccel to identify arrays, so you will also need to look out for this when handling lists